apache/ambari-infra
# AMBARI-24731 - Infra Manager: scheduled cleanup of old jobs

## What changes were proposed in this pull request?

- Create clean up job for deleting expired job data from Infra Manager local database
- Replace aws sdk to minio because aws sdk can not be modularized and it was not worked with localstack which we use for mock s3 in IT
- Refactor job configuration: prior the refactor only job groups could be defined from now single jobs too
- Extract Solr specific functionality from JobScheduler to get a more general scheduler. 
- Use immutable configuration and parameter objects for jobs
- Adjust integration test for the separate infra repo

## How was this patch tested?

1. run Integration test 
2. manually:
- Deploy Ambari and a cluster with infra solr, infra manager, logsearch, hdfs
- enable archive_service_job set the destination to hdfs and run it using infra manager API
- check hdfs contains files in the archives destination directory. Check the content of the files
3. manually:
- start infra manager in a docker environment
- set cleanup schedule to run the job in every minute and set the ttl for a small duration of time
- check infra managers sqlite db contains job data and their creation date
